app/src/pages/inside/testCaseLibraryPage/createTestCaseModal/testCaseUtils.ts

```diff
@@ -73,3 +67,62 @@ export const isDuplicateTestCaseError = (error: unknown) => {
 
   return false;
 };
+
+interface FolderPayload {
+  testFolder?: { name: string; parentTestFolderId?: number | null };
+  testFolderId?: number;
+}
+
+interface ProcessFolderResult {
+  payload: FolderPayload;
+  newFolderDetails: NewFolderData | undefined;
+  existingFolderId: number | undefined;
+}
+
+export const processFolder = (
+  folder: FolderWithFullPath | NewFolderData | string,
+): ProcessFolderResult => {
+  if (isString(folder)) {
+    return {
+      payload: { testFolder: { name: folder } },
+      newFolderDetails: { name: folder, parentTestFolderId: undefined },
+      existingFolderId: undefined,
+    };
+  }
+
+  if (isNewFolderData(folder)) {
+    const parentId = folder.parentTestFolderId;
+
+    return {
+      payload: {
+        testFolder: {
+          name: folder.name,
+          ...(parentId !== undefined && parentId !== 0 && { parentTestFolderId: parentId }),
```

Comment:
What is `parentId !== 0` check for - is it kind of root folder? Maybe let's add variable for it